### ISSUE TITLE
[FORMATTING] SQLite: user-defined functions and table-valued functions have an extra space 

### ISSUE THREAD

**Issue by Prinzhorn** [sql-formatter-org/sql-formatter#444]
It appears there is an internal list of known SQLite functions and virtual tables used to format. I rely heavily on user-defined functions and table-valued functions. These custom functions have an extra space before the opening parenthesis.

**Input data**

```sql
SELECT
  foo('unknown'),
  substr('known'),
  bar('unknown')
FROM
  foo_bar(foo('unknown')),
  json_each()
```

**Expected Output**

```sql
SELECT
  foo('unknown'),
  substr('known'),
  bar('unknown')
FROM
  foo_bar(foo('unknown')),
  json_each()
```

**Actual Output**

```sql
SELECT
  foo ('unknown'), -- Extra space
  substr('known'),
  bar ('unknown') -- Extra space
FROM
  foo_bar (foo ('unknown')), -- Two extra spaces
  json_each()
```

**Usage**

=> https://sql-formatter-org.github.io/sql-formatter/

I don't think any of the formatting options matter in this case. The formatter should not need to keep a list of SQLite functions/tables but infer the formatting from the token (I have no idea how this library is actually implemented).

**Comment by Prinzhorn** [comment]
Thanks for the quick reply. I'm sure this is a tricky problem and you likely don't want to ship like 58MB of grammars https://www.npmjs.com/package/antlr4ts-sql

![image](https://user-images.githubusercontent.com/679144/190858071-6397be13-7df5-432e-955b-3ad215b67580.png)

For my use case I wonder if I could interface to the SQLite C API to tokenize the SQL, but it looks there is no high level API available.

**Comment by ourpower** [comment]
Hey,

Maybe any update on this one behavior?  Or maybe can you suggest any workarounds?

**Comment by nene** [comment]
For SQLite I can suggest a new SQL formatting library that I've been developing: [prettier-plugin-sql-cst](https://github.com/nene/prettier-plugin-sql-cst)

**Comment by karlhorky** [comment]
Yeah this is also happening with PostgreSQL, the `now()` function has an extra space between the function name and the parentheses:

Input:

```sql
SELECT
  users.id,
  users.username
FROM
  users
  INNER JOIN sessions ON (
    sessions.token = 'abc123'
    AND users.id = sessions.user_id
    AND sessions.expiry_timestamp > now()
  )
```

Expected Output (no change):

```sql
SELECT
  users.id,
  users.username
FROM
  users
  INNER JOIN sessions ON (
    sessions.token = 'abc123'
    AND users.id = sessions.user_id
    AND sessions.expiry_timestamp > now()
  )
```

Actual Output (extra space between `now` and `()`):

```sql
SELECT
  users.id,
  users.username
FROM
  users
  INNER JOIN sessions ON (
    sessions.token = 'abc123'
    AND users.id = sessions.user_id
    AND sessions.expiry_timestamp > now ()
  )
```

**Comment by nwalters512** [comment]
@karlhorky did you set `language: 'postgresql'` in your config? `NOW` might not necessary be recognized as a function if you haven't specified the language. For reference, the `NOW` function is handled here:

https://github.com/sql-formatter-org/sql-formatter/blob/51547457a8488092b2e81b536be34dfacbcf5d81/src/languages/postgresql/postgresql.functions.ts#L188

**Comment by karlhorky** [comment]
I believe I do have it configured, yes:

https://github.com/upleveled/eslint-config-upleveled/blob/0f994bc92e1c5731d830a6f738b21b0f62c4fdef/templates/next-js-postgresql/prettier.config.mjs#L15

**Comment by pjxxcc** [comment]
Encountering the same issue, could we consider enumerating scenarios where spaces are needed before parentheses for handling? In all other cases, remove spaces in front of parentheses.

**Comment by pjxxcc** [comment]
Encountering the same issue, can we consider exhaustively listing scenarios where spaces are needed before parentheses for handling? In all other cases, remove spaces in front of parentheses.  
For example, keywords like:  
```where 
when 
and 
join 
as 
+ 
- 
* 
/ 
(
 ```

**Comment by abdukhashimov** [comment]
any update on this thread?

**Comment by nene** [comment]
no
